front/components/agent_builder/settings/TagsSuggestDialog.tsx

```diff
@@ -0,0 +1,172 @@
+  Chip,
+  ContextItem,
+  Dialog,
+  DialogContainer,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  ScrollArea,
+  Spinner,
+  useSendNotification,
+} from "@dust-tt/sparkle";
+import { useEffect, useState } from "react";
+
+import { useAgentConfigurations } from "@app/lib/swr/assistants";
+import { useCreateTag, useTagsSuggestions } from "@app/lib/swr/tags";
+import type { WorkspaceType } from "@app/types";
+
+export const MAX_TAG_LENGTH = 100;
+
+interface TagsSuggestDialogProps {
+  owner: WorkspaceType;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+}
+
+export const TagsSuggestDialog = ({
+  owner,
+  isOpen,
+  setIsOpen,
+}: TagsSuggestDialogProps) => {
+  const { createTag } = useCreateTag({ owner });
+  const sendNotification = useSendNotification();
+  const [appliedSuggestion, setAppliedSuggestion] = useState<
+    Record<string, boolean>
+  >({});
+  const { suggestions, isSuggestionsLoading, isSuggestionsError } =
+    useTagsSuggestions({ owner, disabled: !isOpen });
+
+  const { mutateRegardlessOfQueryParams: mutateAgentConfigurations } =
+    useAgentConfigurations({
+      workspaceId: owner.sId,
+      agentsGetView: "manage",
+      includes: ["authors", "usage", "feedbacks"],
+      disabled: true,
+    });
+
+  const handleCreateTag = async () => {
+    const tags = [];
+    for (const s of suggestions) {
+      if (appliedSuggestion[s.name]) {
+        const tag = await createTag(
+          s.name,
+          s.agents.map((a) => a.sId)
+        );
+        if (tag) {
+          tags.push(tag);
+        }
+      }
+    }
+
+    await mutateAgentConfigurations();
+
+    sendNotification({
+      type: "success",
+      title: "Tag suggestions applied",
+      description: "All tags have been successfully created for your agents.",
+    });
+  };
+
+  useEffect(() => {
+    if (isSuggestionsError) {
+      sendNotification({
+        type: "error",
+        title: "Error getting suggestions",
+        description:
+          "Please try again. If the problem persists, please contact support.",
+      });
+
+      setIsOpen(false);
+    }
+  }, [isSuggestionsError, sendNotification, setIsOpen]);
+
+  useEffect(() => {
+    setAppliedSuggestion(
+      suggestions.reduce(
+        (acc, suggestion) => {
+          acc[suggestion.name] = true;
+          return acc;
+        },
+        {} as Record<string, boolean>
+      )
+    );
+  }, [suggestions]);
+
+  return (
+    <Dialog open={isOpen} onOpenChange={setIsOpen}>
+      <DialogContent size="lg">
+        <DialogHeader>
+          <DialogTitle>Agent tag suggestions</DialogTitle>
+          <DialogDescription>
+            {isSuggestionsLoading ? (
+              <>Analyzing your agents to create tags...</>
+            ) : (
+              <>Tag suggestions from your workspace agents:</>
+            )}
+          </DialogDescription>
+        </DialogHeader>
+        <DialogContainer className="h-[500px]">
```

Comment:
Can we avoid using hardcoded px? You might need to tweak the tailwindconfig though, I don't think we have a `h-125`.